- Description: dasd: ioctl BIODASDDISABLE does not clean up partitions
  Symptom:     A call to BIODASDDISABLE should remove all partition information
               for a dasd. This is typical seen during dasdfmt. When dasdfmt is
               started it calls BIODASDDISABLE and all partition information from
               /sys/block/dasd<x>/ should disappear. Due to this defect the
               partition information is not removed at the beginning of
               formatting, but only by the partition reread at the end of
               formatting.
  Problem:     dasd_destroy_partitions invokes ioctl BLKPG_DEL_PARTITION but this
               is not valid.
  Solution:    The correct ioctl is BLKPG and BLKPG_DEL_PARTITION is an option
               that needs to be given as part of the argument.
  Problem-ID:  --
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/block/dasd_genhd.c	8 Apr 2004 11:09:23 -0000	1.47.2.1
+++ linux-2.5/drivers/s390/block/dasd_genhd.c	18 Jun 2004 10:55:45 -0000	1.47.2.2
@@ -9,7 +9,7 @@
  *
  * gendisk related functions for the dasd driver.
  *
- * $Revision: 1.47.2.1 $
+ * $Revision: 1.47.2.2 $
  */
 
 #include <linux/config.h>
@@ -152,8 +152,9 @@
 	memset(&bpart, sizeof(struct blkpg_partition), 0);
 	memset(&barg, sizeof(struct blkpg_ioctl_arg), 0);
 	barg.data = &bpart;
+	barg.op = BLKPG_DEL_PARTITION;
 	for (bpart.pno = device->gdp->minors - 1; bpart.pno > 0; bpart.pno--)
-		ioctl_by_bdev(bdev, BLKPG_DEL_PARTITION, (unsigned long) &barg);
+		ioctl_by_bdev(bdev, BLKPG, (unsigned long) &barg);
 
 	invalidate_partition(device->gdp, 0);
 	/* Matching blkdev_put to the blkdev_get in dasd_scan_partitions. */

